tests/languages/docker_test.py

```diff
@@ -12,3 +16,75 @@ def invalid_attribute():
         getgid=invalid_attribute,
     ):
         assert docker.get_docker_user() == ()
+
+
+class TestInDocker:
+    @mock.patch('builtins.open', new_callable=mock_open)
+    def test_in_docker_no_file(self, mock_file):
+        mock_file.side_effect = FileNotFoundError
+
+        assert docker.is_in_docker() is False
+        mock_file.assert_called()
+
+    @mock.patch('builtins.open', new_callable=mock_open, read_data='tdockert')
```

Comment:
please prefer the context manager, the decorator form is signature-mutating and much harder to understand the scope and debug.  you'll notice that there is no use of the mock decorator in this project

Reply:
also please use `mock.patch.object` -- `mock.patch` has an undesirable side-effect of triggering an import which can easily lead to test pollution and other strange bugs

Reply:
done